src/frontend/packages/lib_components/src/types/tracks.ts

```diff
@@ -0,0 +1,26 @@
+import { Nullable } from 'lib-common';
+
+/** Base shape for all resources to extend. */
+export interface Resource {
+  id: string;
+}
+
+export interface PlaylistLite extends Resource {
+  title?: string | Nullable<Extract<Playlist, 'title'>>;
+  lti_id?: string | Nullable<Extract<Playlist, 'lti_id'>>;
+}
```

Comment:
Why do we have to write `string | Nullable<Extract<Playlist, 'title'>>`? `Nullable<Extract<Playlist, 'title'>>` wouldn't be enough? Also this kind is quite strange, why don't we directly write:
```suggestion
export type PlaylistLite = Partial<Playlist> & Required<Pick<Playlist, 'id'>>;
```
or
```suggestion
export type PlaylistLite = Partial<Pick<Playlist, 'title' | 'lti_id'>> & Required<Pick<Playlist, 'id'>>;
```